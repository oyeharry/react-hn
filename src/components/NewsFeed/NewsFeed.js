import React from 'react';
import PropTypes from 'prop-types';

import Box from '../Box';
import Text from '../Text';
import Image from '../Image';
import Button from '../Button';
import Anchor from '../Anchor';

function NewsFeed(props) {
  const {
    highlighted,
    voted,
    data,
    onUpVoteButtonClick,
    onHideButtonClick,
  } = props;

  const {
    totalComments,
    totalUpVotes,
    title,
    linkDomain,
    username,
    postedWhen,
    url,
    id,
  } = data;

  return (
    <Box
      display="flex"
      alignItems="center"
      bg={highlighted ? 'blackSqueeze' : 'transparent'}
      flexDirection={['column', 'row']}
      width="100%"
    >
      <Box display="flex" alignItems="center" width="100%">
        <Box ml="2" mr="2" width={['40px', '80px']}>
          <Text textAlign="right">{totalComments}</Text>
        </Box>
        <Box padding="2">
          <Text color="primary.dark">{totalUpVotes}</Text>
        </Box>
        <Box padding="0">
          {!voted && (
            <Button
              padding="0"
              onClick={() => onUpVoteButtonClick(id)}
              aria-label="Up Vote"
            >
              <Image
                alt="Up Vote"
                size="12px"
                src="/assets/images/grayarrow2x.gif"
              ></Image>
            </Button>
          )}
          {voted && <Box size="12px" />}
        </Box>
        <Box
          padding="2"
          display="flex"
          flexDirection={['column', 'column', 'row']}
          alignItems={['flex-start', 'flex-start', 'center']}
        >
          <Box
            padding="0"
            display="flex"
            alignItems={['flex-start', 'center']}
            flexDirection={['column', 'row']}
          >
            <Anchor fontSize={['5', '6']} href={url} p="0">
              <Text
                noTextOverflow
                as="span"
                display="inline-block"
                maxWidth={['200px', '400px', '300px']}
              >
                {title}
              </Text>
            </Anchor>

            <Anchor
              fontSize="4"
              p={['0', '1']}
              mt={['1', '0', '0']}
              color="gray.600"
              href={url}
            >
              <Text as="span">{linkDomain && '('}</Text>
              <Text as="span">{linkDomain}</Text>
              <Text as="span">{linkDomain && ')'}</Text>
            </Anchor>
          </Box>

          <Box display="flex" alignItems="center" mt={['1', '1', '0']}>
            <Text fontSize="4" color="frenchGray" pl={['0', '0', '1']} pr="1">
              by
            </Text>
            <Text fontSize="4">{username}</Text>
          </Box>

          <Box
            mt={['1', '1', '0']}
            pl={['0', '0', '1']}
            display="flex"
            alignItems="center"
          >
            <Text fontSize="4" color="gray.600">
              {postedWhen}
            </Text>
            <Box pl="1" pr="1" display="flex">
              <Button
                padding="0"
                aria-label="Hide News Feed"
                onClick={() => onHideButtonClick(id)}
              >
                <Text fontSize="4" color="gray.600" as="span">
                  [
                </Text>
                <Text fontSize="4" pr="1" pl="1" as="span">
                  hide
                </Text>
                <Text fontSize="4" color="gray.600" as="span">
                  ]
                </Text>
              </Button>
            </Box>
          </Box>
        </Box>
      </Box>
    </Box>
  );
}

NewsFeed.defaultProps = {
  data: {
    totalComments: 0,
    totalUpVotes: 0,
    title: '',
    linkDomain: '',
    username: '',
    postedWhen: '',
    url: '',
    id: '0',
  },
  voted: false,
  highlighted: false,
  onUpVoteButtonClick: () => {},
  onHideButtonClick: () => {},
};

NewsFeed.propTypes = {
  data: PropTypes.shape({
    totalComments: PropTypes.number,
    totalUpVotes: PropTypes.number,
    title: PropTypes.string,
    linkDomain: PropTypes.string,
    username: PropTypes.string,
    postedWhen: PropTypes.string,
    url: PropTypes.string,
    id: PropTypes.string,
  }),
  voted: PropTypes.bool,
  highlighted: PropTypes.bool,
  onUpVoteButtonClick: PropTypes.func,
  onHideButtonClick: PropTypes.func,
};

export default NewsFeed;
