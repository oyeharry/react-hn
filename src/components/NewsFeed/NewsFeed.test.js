import React from 'react';
import renderer from 'react-test-renderer';

import NewsFeed from '.';

const newsFeedData = {
  totalComments: 0,
  totalUpVotes: 0,
  title: '',
  linkDomain: '',
  username: '',
  postedWhen: '',
  url: '',
};

describe('NewsFeed', () => {
  it('Should render component with default props', () => {
    const tree = renderer.create(<NewsFeed data={newsFeedData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render component voted enabled', () => {
    const tree = renderer
      .create(<NewsFeed data={newsFeedData} voted />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
