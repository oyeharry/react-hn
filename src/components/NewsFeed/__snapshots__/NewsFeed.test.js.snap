// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsFeed Should render component voted enabled 1`] = `
<div
  className="sc-AxjAm ccOFVg"
  display="flex"
  width="100%"
>
  <div
    className="sc-AxjAm kzHeKe"
    display="flex"
    width="100%"
  >
    <div
      className="sc-AxjAm bUildF"
      width={
        Array [
          "40px",
          "80px",
        ]
      }
    >
      <p
        className="sc-AxirZ EZtwn"
        color="text.primary"
        fontFamily="primary"
      >
        0
      </p>
    </div>
    <div
      className="sc-AxjAm iAlmwV"
    >
      <p
        className="sc-AxirZ eZcgVC"
        color="primary.dark"
        fontFamily="primary"
      >
        0
      </p>
    </div>
    <div
      className="sc-AxjAm gBtiDd"
    >
      <div
        className="sc-AxjAm yYRaU"
        size="12px"
      />
    </div>
    <div
      className="sc-AxjAm fedZlK"
      display="flex"
    >
      <div
        className="sc-AxjAm iCMFnb"
        display="flex"
      >
        <a
          className="sc-AxhUy eJIopb"
          color="text.primary"
          fontFamily="primary"
          fontSize={
            Array [
              "5",
              "6",
            ]
          }
          href=""
          p="0"
          to=""
        >
          <span
            className="sc-AxirZ byKZYv"
            color="text.primary"
            display="inline-block"
            fontFamily="primary"
          >
            
          </span>
        </a>
        <a
          className="sc-AxhUy iaoSGh"
          color="gray.600"
          fontFamily="primary"
          fontSize="4"
          href=""
          mt={
            Array [
              "1",
              "0",
              "0",
            ]
          }
          p={
            Array [
              "0",
              "1",
            ]
          }
          to=""
        >
          <span
            className="sc-AxirZ igfFwH"
            color="text.primary"
            fontFamily="primary"
          >
            
          </span>
          <span
            className="sc-AxirZ igfFwH"
            color="text.primary"
            fontFamily="primary"
          >
            
          </span>
          <span
            className="sc-AxirZ igfFwH"
            color="text.primary"
            fontFamily="primary"
          >
            
          </span>
        </a>
      </div>
      <div
        className="sc-AxjAm fZrlEf"
        display="flex"
      >
        <p
          className="sc-AxirZ gqgGZh"
          color="frenchGray"
          fontFamily="primary"
          fontSize="4"
        >
          by
        </p>
        <p
          className="sc-AxirZ ePVKpX"
          color="text.primary"
          fontFamily="primary"
          fontSize="4"
        >
          
        </p>
      </div>
      <div
        className="sc-AxjAm jJqNEp"
        display="flex"
      >
        <p
          className="sc-AxirZ gdOnNB"
          color="gray.600"
          fontFamily="primary"
          fontSize="4"
        >
          
        </p>
        <div
          className="sc-AxjAm krOivm"
          display="flex"
        >
          <button
            aria-label="Hide News Feed"
            className="sc-AxhCb lTgye"
            fontFamily="primary"
            fontSize="15px"
            onClick={[Function]}
          >
            <span
              className="sc-AxirZ gdOnNB"
              color="gray.600"
              fontFamily="primary"
              fontSize="4"
            >
              [
            </span>
            <span
              className="sc-AxirZ lnGsdQ"
              color="text.primary"
              fontFamily="primary"
              fontSize="4"
            >
              hide
            </span>
            <span
              className="sc-AxirZ gdOnNB"
              color="gray.600"
              fontFamily="primary"
              fontSize="4"
            >
              ]
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NewsFeed Should render component with default props 1`] = `
<div
  className="sc-AxjAm ccOFVg"
  display="flex"
  width="100%"
>
  <div
    className="sc-AxjAm kzHeKe"
    display="flex"
    width="100%"
  >
    <div
      className="sc-AxjAm bUildF"
      width={
        Array [
          "40px",
          "80px",
        ]
      }
    >
      <p
        className="sc-AxirZ EZtwn"
        color="text.primary"
        fontFamily="primary"
      >
        0
      </p>
    </div>
    <div
      className="sc-AxjAm iAlmwV"
    >
      <p
        className="sc-AxirZ eZcgVC"
        color="primary.dark"
        fontFamily="primary"
      >
        0
      </p>
    </div>
    <div
      className="sc-AxjAm gBtiDd"
    >
      <button
        aria-label="Up Vote"
        className="sc-AxhCb lTgye"
        fontFamily="primary"
        fontSize="15px"
        onClick={[Function]}
      >
        <img
          alt="Up Vote"
          className="sc-AxiKw eYIUPg"
          size="12px"
          src="/assets/images/grayarrow2x.gif"
        />
      </button>
    </div>
    <div
      className="sc-AxjAm fedZlK"
      display="flex"
    >
      <div
        className="sc-AxjAm iCMFnb"
        display="flex"
      >
        <a
          className="sc-AxhUy eJIopb"
          color="text.primary"
          fontFamily="primary"
          fontSize={
            Array [
              "5",
              "6",
            ]
          }
          href=""
          p="0"
          to=""
        >
          <span
            className="sc-AxirZ byKZYv"
            color="text.primary"
            display="inline-block"
            fontFamily="primary"
          >
            
          </span>
        </a>
        <a
          className="sc-AxhUy iaoSGh"
          color="gray.600"
          fontFamily="primary"
          fontSize="4"
          href=""
          mt={
            Array [
              "1",
              "0",
              "0",
            ]
          }
          p={
            Array [
              "0",
              "1",
            ]
          }
          to=""
        >
          <span
            className="sc-AxirZ igfFwH"
            color="text.primary"
            fontFamily="primary"
          >
            
          </span>
          <span
            className="sc-AxirZ igfFwH"
            color="text.primary"
            fontFamily="primary"
          >
            
          </span>
          <span
            className="sc-AxirZ igfFwH"
            color="text.primary"
            fontFamily="primary"
          >
            
          </span>
        </a>
      </div>
      <div
        className="sc-AxjAm fZrlEf"
        display="flex"
      >
        <p
          className="sc-AxirZ gqgGZh"
          color="frenchGray"
          fontFamily="primary"
          fontSize="4"
        >
          by
        </p>
        <p
          className="sc-AxirZ ePVKpX"
          color="text.primary"
          fontFamily="primary"
          fontSize="4"
        >
          
        </p>
      </div>
      <div
        className="sc-AxjAm jJqNEp"
        display="flex"
      >
        <p
          className="sc-AxirZ gdOnNB"
          color="gray.600"
          fontFamily="primary"
          fontSize="4"
        >
          
        </p>
        <div
          className="sc-AxjAm krOivm"
          display="flex"
        >
          <button
            aria-label="Hide News Feed"
            className="sc-AxhCb lTgye"
            fontFamily="primary"
            fontSize="15px"
            onClick={[Function]}
          >
            <span
              className="sc-AxirZ gdOnNB"
              color="gray.600"
              fontFamily="primary"
              fontSize="4"
            >
              [
            </span>
            <span
              className="sc-AxirZ lnGsdQ"
              color="text.primary"
              fontFamily="primary"
              fontSize="4"
            >
              hide
            </span>
            <span
              className="sc-AxirZ gdOnNB"
              color="gray.600"
              fontFamily="primary"
              fontSize="4"
            >
              ]
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
